frontend/src/libs/components/select/select.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  type Control,
+  type FieldErrors,
+  type FieldPath,
+  type FieldValues,
+} from 'react-hook-form';
+import Select from 'react-select';
+
+import { useFormController } from '~/libs/hooks/hooks.js';
+import { type Options } from '~/libs/types/types.js';
+
+import styles from './select.module.scss';
+import { SelectStyles } from './styles.js';
+
+type SelectProperties<T extends FieldValues> = {
+  control: Control<T, null>;
+  name: FieldPath<T>;
+  options: Options<string | number>[];
+  placeholder?: string;
+  errors: FieldErrors<T>;
+};
+
+function findOptionByValue<T extends string | number>(
+  options: Options<T>[],
+  value: T,
+): Options<T> | undefined {
+  return options.find((option) => option.value === value);
+}
+
+const SelectComponent = <T extends FieldValues>({
```

Comment:
```suggestion
const Select = <T extends FieldValues>({
```

Reply:
can i export  to components.ts SelectComponent  as Select,or i need to change import in select.tsx to ReactSelect and rename SelectComponents to Select